main.py

```diff
@@ -44,64 +79,64 @@ def main():
 
     api = kubernetes.client.CoreV1Api()
 
-    device_list = [
-        "gpu",
-        "bme280",
-        "bme680",
-        "gps",
-        "microphone",
-        "raingauge",
-        "airquality",
-    ]
-
     while True:
         logging.info("scanning for devices")
 
-        resources = {device: None for device in device_list}
-
-        iio_names = get_iio_names(args)
-
-        # tag gpu nodes
-        for node in ["nxcore", "nxagent", "sb-core"]:
-            if node in args.kubenode:
-                resources["gpu"] = "true"
-
-        if "bme280" in iio_names:
-            resources["bme280"] = "true"
-
-        if "bme680" in iio_names:
-            resources["bme680"] = "true"
-
-        if Path(args.root, "dev/gps").exists():
-            resources["gps"] = "true"
-
-        if Path(args.root, "dev/airquality").exists():
-            resources["airquality"] = "true"
-
-        lsusb_output = subprocess.check_output(["lsusb", "-v"]).decode()
-
-        if "Microphone" in lsusb_output:
-            resources["microphone"] = "true"
-
-        # NOTE the raingauge uses a generic usb serial connector, so it's hard to tell that it's
-        # specifically the raingauge. we just assume that it's the only one on the rpi
-        if "rpi" in args.kubenode and Path(args.root, "dev/ttyUSB0").exists():
-            resources["raingauge"] = "true"
-
+        # get list of current resources on node and assume not present (will be set below)
+        #  this is ensure that if an item (sensor or capability) currently on the node labels disappears, it is removed on the node update below
+        node_labels = api.read_node(args.kubenode).metadata.labels
+        resources = {i.split(".")[1]: None for i in node_labels if i.split(".")[0] == "resource"}
+
+        # load the node manifest
+        with open(args.manifest) as f:
+            manifest = json.load(f)
+
+        # get the manifest compute dict for this node
+        for compute in manifest["computes"]:
+            if compute["serial_no"] in args.kubenode:
```

Comment:
I would recommend lowering strings in case the serial_no contains any capital latter.

Reply:
good idea. done